### PR TITLE
Merging the code of Polymake/ into Polytopes/

### PR DESCRIPTION
**Pull request by lkastner**
Currently I am simplifying the code of the `Polymake/` folder to use more of the `Polytopes` code. Nevertheless I need input on where the code should end up after refactoring. Not all of it seems to fit perfectly into `Polytopes`.

### CODE REVIEW

File: src/Polymake/Ineq.jl [hunk 1]
```diff
@@ -31,121 +31,56 @@ function _polytope(; A::fmpz_mat=zero_matrix(FlintZZ, 1, 1), b::fmpz_mat=zero_ma
...
-    end
-  end
-  for i=1:nrows(inner)
-    @assert inner[i,1] == 1
-    for j=1:ncols(A)
-      res[i+nrows(out), j] = inner[i, j+1]
-    end
-  end
-  return res
+  return _polytope_new(A = A, b = b, C = identity_matrix(FlintZZ, ncols(A)))
```
Comment (fieker) [inline src/Polymake/Ineq.jl]: Are you sure this returns solutions? It looks like the the polytope itself is returned here...

File: src/Polymake/Ineq.jl [hunk 2]
```diff
@@ -31,121 +31,56 @@ function _polytope(; A::fmpz_mat=zero_matrix(FlintZZ, 1, 1), b::fmpz_mat=zero_ma
...
 
-function _cone(; A::fmpz_mat=zero_matrix(FlintZZ, 1, 1), b::fmpz_mat=zero_matrix(FlintZZ, ncols(A), 1), C::fmpz_mat=zero_matrix(FlintZZ, 1, 1))
-  if !iszero(A)
-    bA = Matrix{BigInt}(hcat(-b, A))
-    z = findall(i->!iszero_row(bA, i), 1:nrows(bA))
-    zbA = Matrix{BigInt}(bA[z, :])
-  else
-    zbA = Matrix{BigInt}(undef, 0, 0)
-  end
-  if !iszero(C)
```
Comment (fieker) [inline src/Polymake/Ineq.jl:42]: are you (PM) able to handle zero rows?

**Comment by fieker** [comment]
to exercise some of the code:
```julia
k, a = quadratic_field(10)
zk = maximal_order(k)
factorisations(15*zk(a))
irreducibles(collect(keys(factor(15*zk))))
norm_equation(zk, 6)
```
by changing the `15` and `6` you can make polymakes life harder...

**Comment by fieker** [comment]
I have absolutely no opinion about the final resting place of this code snippet. `git grep` and `methods` will find it!

File: src/Polymake/Ineq.jl (hunk 1 above)
Comment (lkastner) [reply src/Polymake/Ineq.jl]: This is just a temporary name. I will change it to something better. I used this to replace `_polytope` in some places, but it did not work out in all places yet.

File: src/Polymake/Ineq.jl (hunk 2 above)
Comment (lkastner) [reply src/Polymake/Ineq.jl:42]: Yes, I actually built this in just now for this occasion.

**Comment by lkastner** [comment]
I moved the refactored `Ineq.jl` into the `Polytopes` folder. As for the `NmbThy.jl`: I renamed the `Polymake` folder to `NumberTheory`. This seemed to make more sense given the name of the file. There is still the following comment:
```
  #solving Ax = b for x >=0  and A >=0 implies that colums of A with
  #entries > those in b can never be part of a solution
  #thus we prune them
  #of course, this could/ should be done in solve_non_negative...
  #hence in polymake
 ```
 I think `solve_non_negative` is doing the right thing now, but it was unclear to me what needs to be replaced in the function containing that comment. Also there was no documentation in `NmbThy.jl`, so I did not want to mess with it too much.

Nevertheless, I added documentation and examples to the `Ineq.jl`. So I think we can merge this now and then further refactor later.

**Comment by lkastner** [comment]
I don't know why the docs fail.

**Comment by thofma** [comment]
One signature for `solve_mixed` is included twice.

**Comment by thofma** [comment]
One signature for `solve_mixed` is included twice.

File: src/Polytopes/Polyhedron/properties.jl [hunk 3]
```diff
@@ -420,6 +420,64 @@ function lattice_points(P::Polyhedron)
...
+ [0, 0, 0]
+```
+"""
+function interior_lattice_points(P::Polyhedron)
+    if pm_polytope(P).BOUNDED
+        lat_pts = pm_polytope(P).INTERIOR_LATTICE_POINTS
+        return VectorIterator{PointVector{Polymake.Integer}}(lat_pts[:, 2:end])
+    else
+        throw(ArgumentError("Polyhedron not bounded"))
+    end
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/properties.jl]: Normally I would recommend to validate arguments at the start; this leads to less indentation and IMHO makes it easier to follow the code. Of course in trivial cases like this, it's fine either way. So this is a purely optional suggestion
```suggestion
    pm_polytope(P).BOUNDED || throw(ArgumentError("Polyhedron not bounded"))
    lat_pts = pm_polytope(P).INTERIOR_LATTICE_POINTS
    return VectorIterator{PointVector{Polymake.Integer}}(lat_pts[:, 2:end])
```

File: src/Polytopes/Polyhedron/properties.jl [hunk 4]
```diff
@@ -420,6 +420,64 @@ function lattice_points(P::Polyhedron)
...
+ [-1, 0, 0]
+ [0, -1, 0]
+ [0, 0, -1]
+ [0, 0, 1]
+ [0, 1, 0]
+ [1, 0, 0]
+```
+"""
+function boundary_lattice_points(P::Polyhedron)
+    if pm_polytope(P).BOUNDED
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/properties.jl]: Same remark as above

**Review by fingolfin — APPROVED** [review]
Looks good to me; minor optional remarks added

**Comment by lkastner** [comment]
> One signature for `solve_mixed` is included twice.

Thank you. For some reason I did not understand this error message properly.

**Pull request merged**